packages/react/src/primitives/types/view.ts

```diff
@@ -18,15 +32,32 @@ export type PrimitiveProps<
   Props extends ViewProps,
   Element extends ElementType
 > = MergeProps<
-  Omit<Props, 'as'> & { as?: Element | Props['as'] },
-  ElementProps<Element>
+  Omit<Props, 'as'> & {
+    as?: Element | Props['as'];
+  },
+  Omit<ElementProps<Element>, 'ref'> // exclude `ref?: LegacyRef` included in DetailedHTMLProps
```

Comment:
This Omit is required because `JSX.IntrinsicElements[Element]` resolves to React.DetailedHTMLProps, which resolves to `ClassAttributes`, which contains the `ref?: LegacyRef<T>`. AFAIK, it's invalid to include the `ref` prop in a functional component, and causes type errors on functional component primitives. For example, `Text` throws a type error when passing props to View because `React.LegacyRef` and `React.Ref` are not compatible.

Reply:
🙌🏽